Polishing README; change names of engine and account profile params (#18)
firebolt-db/dbt-firebolt: README.md

#### `quote_columns`

To prevent a warning, you should add a configuration as below to your `dbt_project.yml`. For more info, see the [relevant dbt docs page](https://docs.getdbt.com/reference/resource-configs/quote_columns)

```yml
seeds:
  +quote_columns: false  #or `true` if you have csv column headers with spaces
```

#### Supporting Concurrent Development

In dbt, database schemas are used to compartmentalize developer environments so that concurrent development does not cause table name collisions. Firebolt, however, does not currently support database schemas (it is on the roadmap). To work around this, we recommend that you add the following macro to your project. This macro will take the `schema` field of your `profiles.yml` file and use it as a table name prefix.

```sql
-- macros/generate_alias_name.sql
{% macro generate_alias_name(custom_alias_name=none, node=none) -%}
    {%- if custom_alias_name is none -%}
        {{ node.schema }}__{{ node.name }}
    {%- else -%}
        {{ node.schema }}__{{ custom_alias_name | trim }}
    {%- endif -%}
{%- endmacro %}
```

For an example of how this works, let’s say Shahar and Eric are both working on the same project.

> In their `.dbt/profiles.yml`, Sharar sets `schema=sh`, whereas Eric sets `schema=er`. When they each run the `customers` model, the models will land in the database as tables named `sh_customers` and `er_customers`, respectively. When running dbt in production, you would use yet another `profiles.yml` with nothing set in the `schema` field (or any string of your choice).

Reviewer: ```suggestion
In her `.dbt/profiles.yml`, Sharar sets `schema=sh`, whereas Eric sets `schema=er` in his. When each runs the `customers` model, the models will land in the database as tables named `sh_customers` and `er_customers`, respectively. When running dbt in production, you would use yet another `profiles.yml` with nothing set in the `schema` field (or any string of your choice).
```

Reviewer: I don't understand the last sentence. Why would it be different in production? And we said above that `schema` was a necessary field, but now I think you're saying you can set it as nothing.

Author: Perhaps a fact-check is needed here - is `schema` actually required in the profile? What would I do if don't want a prefix on prod?

Reviewer: It's a required field. dbt will refuse to run it without it. It also can't be blank; I tried.

| Partitioned tables           | :x: |

## Using dbt-firebolt

### Model Configuration for Fact Tables

A dbt model can be created as a Firebolt fact table and configured using the following syntax:

Project file syntax
```yml
models:
  <resource-path>:
    +materialized: table
    +table_type: fact
    +primary_index: [ <column-name>, ... ]
    +indexes:
      - type: aggregating
        key_column: [ <column-name>, ... ]
        aggregation: [ <agg-sql>, ... ]
      ...
```

Property file syntax
```yml
models:
  - name: <model-name>
    config:
      materialized: table
      table_type: fact
      primary_index: [ <column-name>, ... ]
      indexes:
        - type: aggregating | join
          key_column: [ <column-name>, ... ]
          aggregation: [ <agg-sql>, ... ]
        ...
```

Config block syntax
```
{{ config(
    materialized = "table"
    table_type = "fact"
    primary_index = [ "<column-name>", ... ],
    indexes = [
      {
        type = "aggregating"
        key_column = [ "<column-name>", ... ],
        aggregation = [ "<agg-sql>", ... ],
      },
      ...
    ]
) }}
```

#### Description of Configurations Pertaining to Fact Tables

> | Configuration     | Description                                                                               |

Author: I checked the lingo in dbt docs and "configuration" is the language they use to describe each field in a model config. See https://docs.getdbt.com/reference/configs-and-properties

|-------------------|-------------------------------------------------------------------------------------------|

> | `materialized`    | How the model will be materialized into Firebolt. Must be `table` to create a fact table. |

Reviewer: Isn't it the case that it must be `table` period? I thought for now we weren't supporting `view`?

Author: AFAIK we do support ephemeral materializations as well. They are manifested as CTEs in Firebolt. @swanderz fact check please

| `aggregation`     | Sets the aggregations on the aggregating index using the inputted list of SQL agg expressions. |

#### Example of a Fact Table With an Aggregating Index (Config Block Syntax)

```
{{ config(
    materialized = "table",
    table_type = "fact",
    primary_index = "id",
    indexes = [
      {
        type: "aggregating",
        key_column: "order_id",
        aggregation: ["COUNT(DISTINCT status)", "AVG(customer_id)"]
      }
    ]
) }}
```

### Model Configuration for Dimension Tables

A dbt model can be created as a Firebolt dimension table and configured using the following syntax:

Project file syntax
```yml
models:
  <resource-path>:
    +materialized: table
    +table_type: dimension
    +indexes:
      - type: join
        join_column: <column-name>
        dimension_column: [ <column-name>, ... ]
      ...
```

Property file syntax
```yml
models:
  - name: <model-name>
    config:
      materialized: table
      table_type: dimension
      indexes:
        - type: join
          join_column: <column-name>
          dimension_column: [ <column-name>, ... ]
        ...
```

Config block syntax
```
{{ config(

>     materialized = "table"

Reviewer: See my question above.

    table_type = "dimension",
    indexes = [
      {
        type = "join",
        join_column = "<column-name>",
        dimension_column: [ "<column-name>", ... ]
      },
      ...
    ]
) }}
```

#### Description of Configurations Pertaining to Dimension Tables

| Configuration      | Description                                                                               |
|--------------------|-------------------------------------------------------------------------------------------|

> | `materialized`     | How the model will be materialized into Firebolt. Must be `table` to create a dimension table. |

Reviewer: etc.

### Managing Ingestion via External Tables

`dbt-firebolt` supports dbt's [external tables feature](https://docs.getdbt.com/reference/resource-properties/external), which allows dbt to manage the table ingestion process from S3 into Firebolt. This is an optional feature but can be highly convenient depending on your use case.

More information on using external tables including properly configuring IAM can be found in the Firebolt [documentation](https://docs.firebolt.io/sql-reference/commands/ddl-commands#create-external-table).

#### Installation of External Tables Package

To install and use `dbt-external-tables` with Firebolt, you must:

1. Add this package to your packages.yml:
    ```yml
    packages:
    - package: dbt-labs/dbt_external_tables
        version: <version>
    ```
2. Add these fields to your `dbt_project.yml`: 
    ```yml
    dispatch:
      - macro_namespace: dbt_external_tables
        search_order: ['dbt', 'dbt_external_tables']
    ```
3. Pull in the `packages.yml` dependencies by calling `dbt deps`.

#### Using External Tables

> To use external tables, you must define a table as `external` in your `dbt_project.yml` file. Every external table must contain the fields `url`, `type`, and `object_pattern`. Note that the Firebolt external table specifications require fewer fields than what is specified in the dbt documentation.

Reviewer: ```suggestion
To use external tables, you must define a table as `external` in your `dbt_project.yml` file. Every external table must contain the fields `url`, `type`, and `object_pattern`. Note that the Firebolt external table specification requires fewer fields than what is specified in the dbt documentation.
```
